### PR TITLE
[READY] [SQL CHANGE] P A R A L L A X

### PR DESCRIPTION
**Pull request by AffectedArc07**
**What does this PR do:**
Adds Parralax, with 4 modes. This is incomplete, and I would like assistance if possible

- [x] Parallax for on station people (99% of the users)
- [x] Code to save the Parallax (SQL CHANGE REQUIRED)
- [x] All Parallax intensities (Off, Low, Medium, High (Default), Insane)
- [x] Parallax working for ghosts (Waiting for ghost HUD to be merged)
- [ ] Parallax for shuttle hyperspace (Probably needs shuttle subsystem rework)

Currently we still use the same hyperspace:
V5MjZWyFqB
This will be **VERY** hard to implement due to us not having dynamic shuttle area creation (Which TG has). An overhaul of the shuttle subsystem is probably in order, and that is **WAY** out of the scope of this PR

**Images of sprite/map changes:**
Parallax in action: http://affectedarc07.co.uk/CDN/Public/parallax2.gif
(Hosted on own server because git doesnt let me use stuff bigger than 10MB)
image

# PERFORMANCE
To all those, commenting on performance issues, this section is for you. This uses SSparallax, the same method TG uses. I have tested this with 4 people on, and we had no issues with it. A testmerge for this is requested but this doesnt have the same issues that the old system did. If anyone would like a demo of the parallax, either download my fork or hit me up on discord and I can get this hosted for you to try

**Changelog:**
:cl: AffectedArc07
add: P A R R A L A X (That fancy space thing)
/:cl:

### CODE REVIEW

**Comment by farie82** [comment]
that does look fancy. Real fancy

**Comment by FalseIncarnate** [comment]
We've tried and reverted parallax before (https://github.com/ParadiseSS13/Paradise/pull/5235) because it was horrible in terms of processing cost and frankly just looks bad when paired with everything else.

Your version looks marginally better, but still is a pretty huge difference from the rest of the graphics. The gif you posted doesn't make it look much smoother, though that may be the gif being laggy rather than what you were recording.

Honestly don't think we need parallax, and that it frankly is going to just be something that most people will disable for one reason or another.

**Comment by KasparoVy** [comment]
I remember the last time we merged parallax, we had a pretty big performance hit regarding input lag.

I loved it but the input lag was rough

**Comment by AffectedArc07** [comment]
@KasparoVy @FalseIncarnate Re-read OP

**Comment by KasparoVy** [comment]
Hopefully SSParallax holds up just as well with 4*25 people

**Comment by AffectedArc07** [comment]
OP has been updated with a better gif of the parallax, just waiting for CF to cache it properly

**Comment by craftxbox** [comment]
use mp4s, gifs are really bad for compression

**Comment by KasparoVy** [comment]
You better believe I'm putting parallax on insane

**Comment by taukausanake** [comment]
> I'm going to really REALLY miss the way our classic spess tiles look

This. While I really want parallax to work I also really don't want to lose the old space tiles. They look special and unique.

In addition, I don't think this parallax is nearly as impressive as others, namely #4739. Now _that_ was some pretty spess. If we could get that again minus the performance hits I would be all for it

**Comment by Dinarzad** [comment]
> Honestly don't think we need parallax, and that it frankly is going to just be something that most people will disable for one reason or another.

We don't need 80% of the features we have already, we have them because they are nice.  We don't need tattoos/body markings or clothing loadouts or different prosthetic manufacturers, A lot of people ignore those too, but we have them because they look nice aesthetically.
If the changes Arc has done here do fix those performance issues, then I see no reason why not. Especially since it can be turned off. If the performance hits are minimal as stated, then there's literally no reason not to do it.

> This. While I really want parallax to work I also really don't want to lose the old space tiles. They look special and unique.

Bear in mind there's also that space carpeting you can get that could still use the old space tiles as a way to keep both.

**Comment by Markolie** [comment]
This also opens the door to having awesome planets in the background of the station (https://user-images.githubusercontent.com/20987591/37424700-72ef9a8a-27c1-11e8-98f8-2c8c748d321e.png). That's even more reasons to add this!

**Comment by Fox-McCloud** [comment]
I put in this request for this PR.

The system is astoundingly beautiful, and the problem we had with it was never the system itself, so much as its performance impact.

This has been on TG for a couple of years now, and has been through multiple phases of fixes, bugtesting, and performance alterations. They don't have the problems with it we did, and I don't really expect we'll have many problems with it anymore, either.

**Comment by KasparoVy** [comment]
Beautiful planets

**Comment by AffectedArc07** [comment]
If you actually read the OP, you would have seen there is an option to fully disable the parallax, if you like the current space. As for the parallax images, I have two choices. One that closer resembles our current space, and one that looks beautiful (like now)

**Comment by AffectedArc07** [comment]
Right opinion time. These links are not going to be fully accurate of what the parallax looks like ingame, but they will be a basic idea

React with the following one that you agree on:
👀 https://affectedarc07.github.io/SS13WebMap/Paradise/Cyberiad/index.html

🚀 https://affectedarc07.github.io/SS13WebMap/TG/BoxStation/index.html

(Just scroll the station far away and stare)

**Comment by KasparoVy** [comment]
Gah, both parallaxes are absolutely stunning! I love the old-style one b/c nostalgia etc. etc. but newspess looks awesome as well.

I'm super hung up on the art style change but newspess goes hand in hand with all those other featurettes like planets, etc.

Here's some spess wallpapers I captured from your site:
para ss13 old space wallpaper
para ss13 new space wallpaper

**Comment by AffectedArc07** [comment]
Off topic but I have a wallpaper engine skin that’s the space parallax if you really want the wallpaper animated

**Comment by KasparoVy** [comment]
I'd love it, that'd be fantastic

**Comment by Twinmold93** [comment]
Now we can start figuring out a new science of cross referencing the positions of stars to figure out where we are in the vastness of space without needing a GPS (which would still, of course, be easier).

In seriousness, how does the crossing over of z-levels work for this?

**Comment by Tayyyyyyy** [comment]
I love this!! Can't wait for parallax shuttles!

File: paradise.dme [hunk 1]
```diff
@@ -222,9 +223,10 @@
 #include "code\controllers\subsystem\nanoui.dm"
 #include "code\controllers\subsystem\nightshift.dm"
 #include "code\controllers\subsystem\npcai.dm"
-#include "code\controllers\subsystem\npcpool.dm"
+#include "code\controllers\subsystem\npcpool.dm
```
Comment (AffectedArc07) [inline paradise.dme]: This didnt happen

**Comment by AffectedArc07** [comment]
Parallax now works for ghosts, only thing that doesnt work now is shuttle parallax, which is probably out of the scope of this PR due to how our currently shuttlecode works. I would still hold off on merging though because an SQL change mid migration seems like a bad idea

**Comment by KasparoVy** [comment]
If you'd be down to hook us up with your parallax WP engine skins that'd be amazing @AffectedArc07. I'm super interested.

**Comment by AffectedArc07** [comment]
@KasparoVy Can we talk this on discord? This thread is getting off topic as it is

**Comment by AffectedArc07** [comment]
@Fox-McCloud this needs `SQL Change`

**Comment by AffectedArc07** [comment]
Im not reworking shuttle code in this PR

**Comment by AffectedArc07** [comment]
Gonna see if I can make appveyor kick in

**Comment by AffectedArc07** [comment]
Take 2

**Comment by FreeStylaLT** [comment]
those are some close ass stars to be parallaxing like that

**Comment by AffectedArc07** [comment]
@FreeStylaLT quiet you

**Comment by AffectedArc07** [comment]
Deconflicted

**Pull request closed without merging**